src/main/cpp/jsonlayout.cpp

```diff
@@ -0,0 +1,273 @@
+	output.append(": ");
+	appendQuotedEscapedString(output, event->getMessage());
+
+	appendSerializedMDC(output, event);
+
+	appendSerializedNDC(output, event);
+
+
+	if (locationInfo)
+	{
+		output.append(", ");
+		appendQuotedEscapedString(output, "location_info");
+		output.append(": { ");
+		const LocationInfo& locInfo = event->getLocationInformation();
+
+		appendQuotedEscapedString(output, "file");
+		output.append(": ");
+		LOG4CXX_DECODE_CHAR(fileName, locInfo.getFileName());
+		appendQuotedEscapedString(output, fileName);
+		output.append(", ");
+
+		appendQuotedEscapedString(output, "line");
+		output.append(": ");
+		LogString lineNumber;
+		StringHelper::toString(locInfo.getLineNumber(), p, lineNumber);
+		appendQuotedEscapedString(output, lineNumber);
+		output.append(", ");
+
+		appendQuotedEscapedString(output, "class");
+		output.append(": ");
+		appendQuotedEscapedString(output, locInfo.getClassName());
+		output.append(", ");
+
+		appendQuotedEscapedString(output, "method");
+		output.append(": ");
+		appendQuotedEscapedString(output, locInfo.getMethodName());
+
+		output.append(" } ");
+	}
+
+	output.append(" }");
+	output.append(LOG4CXX_EOL);
+
+}
+
+void JSONLayout::appendQuotedEscapedString(LogString& buf,
+	const LogString& input) const
+{
+	/* add leading quote */
+	buf.push_back(0x22);
+
+	logchar specialChars[] =
+	{
+		0x08,   /* \b backspace         */
+		0x09,   /* \t tab               */
+		0x0a,   /* \n newline           */
+		0x0c,   /* \f form feed         */
+		0x0d,   /* \r carriage return   */
+		0x22,   /* \" double quote      */
+		0x5c
+	}; /* \\ backslash         */
+
+
+	size_t start = 0;
+	size_t found = input.find_first_of(specialChars, start);
+
+	while (found != LogString::npos)
+	{
+		if (found > start)
+		{
+			buf.append(input, start, found - start);
+		}
+
+		switch (input[found])
+		{
+			case 0x08:
+				/* \b backspace */
+				buf.push_back(0x5c);
+				buf.push_back('b');
+				break;
+
+			case 0x09:
+				/* \t tab */
+				buf.push_back(0x5c);
+				buf.push_back('t');
+				break;
+
+			case 0x0a:
+				/* \n newline */
+				buf.push_back(0x5c);
+				buf.push_back('n');
+				break;
+
+			case 0x0c:
+				/* \f form feed */
+				buf.push_back(0x5c);
+				buf.push_back('f');
+				break;
+
+			case 0x0d:
+				/* \r carriage return */
+				buf.push_back(0x5c);
+				buf.push_back('r');
+				break;
+
+			case 0x22:
+				/* \" double quote */
+				buf.push_back(0x5c);
+				buf.push_back(0x22);
+				break;
+
+			case 0x5c:
+				/* \\ backslash */
+				buf.push_back(0x5c);
+				buf.push_back(0x5c);
+				break;
+
+			default:
+				buf.push_back(input[found]);
+				break;
+		}
+
+		start = found + 1;
+
+		if (found < input.size())
+		{
+			found = input.find_first_of(specialChars, start);
+		}
+		else
+		{
+			found = LogString::npos;
+		}
+	}
+
+	if (start < input.size())
+	{
+		buf.append(input, start, input.size() - start);
+	}
+
+	/* add trailing quote */
+	buf.push_back(0x22);
+
+	return;
```

Comment:
You are mostly not using empty `return`s, so I suggest removing the current once.

Reply:
Fixed.